packages/filebrowser/src/listing.ts

```diff
@@ -2475,6 +2475,10 @@ export class DirListing extends Widget {
         this.selection[path] = true;
       }
     }
+    // Emit the signal once for all items that were selected.
+    if (existing.length > 0) {
+      this._selectionChanged.emit();
```

Comment:
Should it check if `existing != this.selection`? If I read it right the change could be due to a file/folder being removed in a model update, but a lot of the time the code above would effectively be a no-op, right?

Reply:
I think the main issue is the `this.clearSelectedItems();` call above which emits the `selectionChanged`. So the signal is first emitted, the selection is then empty because `this.clearSelectedItems()` clears the selection, so the signal has to be emitted again after the selection is restored?

Reply:
Could a solution be adding a new argument and calling it like `clearSelectedItems(emit=false)`?